tests/dag_processing/test_dag_bundles.py

```diff
@@ -261,7 +268,53 @@ def test_subdir(self, git_repo):
             tracking_ref=GIT_DEFAULT_BRANCH,
             subdir=subdir,
         )
+        bundle.initialize()
 
         files_in_repo = {f.name for f in bundle.path.iterdir() if f.is_file()}
         assert str(bundle.path).endswith(subdir)
         assert {"some_new_file.py"} == files_in_repo
+
+    @mock.patch("airflow.providers.ssh.hooks.ssh.SSHHook")
+    @mock.patch("airflow.dag_processing.bundles.git.Repo")
+    def test_with_ssh_conn_id(self, mock_gitRepo, mock_hook):
+        repo_url = "git@github.com:apache/airflow.git"
+        conn_id = "ssh_default"
+        bundle = GitDagBundle(
+            repo_url=repo_url,
+            name="test",
+            refresh_interval=300,
+            ssh_conn_kwargs={"ssh_conn_id": "ssh_default"},
+            tracking_ref=GIT_DEFAULT_BRANCH,
+        )
+        bundle.initialize()
+        mock_hook.assert_called_once_with(ssh_conn_id=conn_id)
+
+    @mock.patch("airflow.providers.ssh.hooks.ssh.SSHHook")
+    @mock.patch("airflow.dag_processing.bundles.git.Repo")
+    def test_refresh_with_ssh_connection(self, mock_gitRepo, mock_hook):
+        repo_url = "git@github.com:apache/airflow.git"
+        bundle = GitDagBundle(
+            repo_url=repo_url,
+            name="test",
+            refresh_interval=300,
+            ssh_conn_kwargs={"ssh_conn_id": "ssh_default"},
+            tracking_ref=GIT_DEFAULT_BRANCH,
+        )
+        bundle.initialize()
+        bundle.refresh()
+        # check remotes called twice. one at initialize and one at refresh above
+        assert mock_gitRepo.return_value.remotes.origin.fetch.call_count == 2
+
+    def test_repo_url_starts_with_git_when_using_ssh_conn_id(self):
+        repo_url = "https://github.com/apache/airflow"
```

Comment:
```suggestion
    @pytest.mark.parametrize(
        "repo_url",
        [
            pytest.param("https://github.com/apache/airflow", id="https_url"),
            pytest.param("airflow@example:apache/airflow.git", id="does_not_start_with_git_at"),
            pytest.param("git@example:apache/airflow", id="does_not_end_with_dot_git"),
        ]
    )
    def test_repo_url_starts_with_git_when_using_ssh_conn_id(self, repo_url):
```

Or similar. This better tests that we enforce it both starts with and ends with git.